File: Leetcode/563-Binary-Tree-Tilt/BinaryTreeTilt.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

var tilt int

func findTilt(root *TreeNode) int {
	tilt = 0
	postOrder(root)
	return tilt
}

func postOrder(node *TreeNode) int {
	if node == nil {
		return 0
	}
	left := postOrder(node.Left)
	right := postOrder(node.Right)
	tilt += abs(left - right)
	return left + right + node.Val
}

func abs(i int) int {
	if i < 0 {
		return -i
	}
	return i
}

File: Leetcode/563-Binary-Tree-Tilt/BinaryTreeTilt.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int findTilt(TreeNode* root) {
        postOrder(root);
        return tilt;
    }
private:
    int tilt = 0;

    int postOrder(TreeNode* node) {
        if (node == NULL) {
            return 0;
        }
        int left = postOrder(node->left);
        int right = postOrder(node->right);
        tilt += abs(left - right);
        return left + right + node->val;
    }
};
